os_migrate/plugins/module_utils/server.py

```diff
@@ -128,9 +141,60 @@ def update_migration_params(self, params_dict):
             if value not in choices:
                 raise exc.UnexpectedChoice('floating_ip_mode', choices, value)
 
+        # TODO: add some validation for block device mapping v2
+        if 'data_copy' in params_dict:
+            value = params_dict['data_copy']
+            if not isinstance(value, bool):
+                raise exc.InvalidInputType('data_copy', 'boolean')
+
         return super().update_migration_params(params_dict)
 
-    def update_sdk_params_block_device_mapping(self, sdk_params, block_device_mapping):
+    def update_sdk_params_block_device_mapping_nocopy(self, sdk_params):
+        params, info = self.params_and_info()
+        migration_params = self.migration_params()
+        sdk_params['block_device_mapping'] = []
+        block_device_mapping = sdk_params['block_device_mapping']
+        image_id = sdk_params.get('image_id', None)
+        additional_volumes = migration_params['additional_volumes']
+
+        # Update block device mapping with volumes if data_copy is false
+        if migration_params['boot_volume']['uuid']:
+            boot_vol_mapping = {
+                "boot_index": 0,
+                "uuid": migration_params['boot_volume']['uuid'],
+                'delete_on_termination': False,
+                'destination_type': 'volume',
+                'source_type': 'volume',
+            }
+            block_device_mapping.append(boot_vol_mapping)
+            sdk_params.pop("image_id", None)
+        elif image_id is not None:
```

Comment:
I think this is still None because an earlier piece of logic for copying the boot disks is kicking in here and setting the image_id to None, here:

https://github.com/os-migrate/os-migrate/blob/ba207d3eb909d075b453d6c2a86fe58498c3c7e4/os_migrate/plugins/module_utils/server.py#L306

We should be setting the image_id to none only if we are doing data_copy in general *and* specifically if we're doing also boot_disk_copy. So i think the condition there should be:

```
if self.migration_params()['data_copy'] and self.migration_params()['boot_disk_copy']:
```